src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -613,35 +613,34 @@ private static TypeWithAnnotations DelegateReturnTypeWithAnnotations(MethodSymbo
             var parameterRefKindsBuilder = ArrayBuilder<RefKind>.GetInstance(ParameterCount);
             var parameterScopesBuilder = ArrayBuilder<DeclarationScope>.GetInstance(ParameterCount);
             var parameterTypesBuilder = ArrayBuilder<TypeWithAnnotations>.GetInstance(ParameterCount);
+            bool getEffectiveScopeFromSymbol = false;
+
             for (int i = 0; i < ParameterCount; i++)
             {
                 var refKind = RefKind(i);
-                var scope = Scope(i);
+                var scope = DeclaredScope(i);
                 var type = ParameterTypeWithAnnotations(i);
                 if (scope == DeclarationScope.Unscoped &&
                     ParameterHelpers.IsRefScopedByDefault(Binder.UseUpdatedEscapeRules, refKind, type))
                 {
                     scope = DeclarationScope.RefScoped;
+                    if (_unboundLambda.ParameterAttributes(i).Any())
+                    {
+                        getEffectiveScopeFromSymbol = true;
+                    }
                 }
                 parameterRefKindsBuilder.Add(refKind);
                 parameterScopesBuilder.Add(scope);
                 parameterTypesBuilder.Add(type);
             }
             var parameterRefKinds = parameterRefKindsBuilder.ToImmutableAndFree();
-            var parameterScopes = parameterScopesBuilder.ToImmutableAndFree();
             var parameterTypes = parameterTypesBuilder.ToImmutableAndFree();
 
+            LambdaSymbol? lambdaSymbol = null;
+
             if (!HasExplicitReturnType(out var returnRefKind, out var returnType))
             {
-                var lambdaSymbol = new LambdaSymbol(
-                    Binder,
-                    Binder.Compilation,
-                    Binder.ContainingMemberOrLambda,
-                    _unboundLambda,
-                    parameterTypes,
-                    parameterRefKinds,
-                    refKind: default,
-                    returnType: default);
+                lambdaSymbol = createLambdaSymbol();
```

Comment:
It seems strange that we will sometimes create a lambda symbol which gets thrown away in order to produce information like the inferred return type or effective scope of parameters, and then later on presumably produce a "final" lambda symbol. Is this a sign that there's some design debt here which could be addressed in the future?

Reply:
> Is this a sign that there's some design debt here which could be addressed in the future?

I do not think so. As long as this doesn't become a source of performance problem, that isn't something that I would worry about.